src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java

```diff
@@ -0,0 +1,616 @@
+
+					// perform the request with the given parameters
+					httpResponse = HttpUtil.post(fullRequestUri, body,
+							contentType);
+				} else {
+					// perform the request with the given parameters
+					httpResponse = HttpUtil.post(fullRequestUri, queryParams);
+				}
+
+			} else {
+				// otherwise throw an exception
+				throw new InterceptorException("Only GET or POST method "
+						+ "is allowed");
+			}
+
+		} catch (URISyntaxException | UnsupportedEncodingException e) {
+			LOG.error("Error while sending request: " + e.getMessage());
+		}
+
+		return httpResponse;
+
+	}
+
+	/**
+	 * @return
+	 * @throws UnsupportedEncodingException
+	 *
+	 */
+	private static HttpHeaders getHeadersToForward(HttpHeaders headers)
+			throws UnsupportedEncodingException {
+
+		HttpHeaders responseHeaders = new HttpHeaders();
+
+		if (headers == null) {
+			LOG.debug("No headers found to forward!");
+			return responseHeaders;
+		}
+
+		LOG.debug("Requested to filter the Headers to respond with:");
+
+		for (Entry<String, List<String>> header : headers.entrySet()) {
+			String headerKey = header.getKey();
+			String headerVal = StringUtils.join(header.getValue(), ",");
+
+			LOG.debug("  * Header: " + headerKey);
+
+			if (Arrays.asList(FORWARD_HEADER_KEYS).contains(headerKey)) {
+
+				// the GeoServer response may contain a subtype in the
+				// "Content-Type" header without double quotes surrounding the
+				// subtype's value. If this is set we need to surround it
+				// with double quotes as this is required by the Spring
+				// ResponseEntity (and the RFC 2616 standard).
+				Pattern pattern = Pattern.compile("subtype=(.*)");
+				Matcher matcher = pattern.matcher(headerVal);
+
+				if (matcher.find()) {
+					String replaceCandidate = matcher.group(1);
+					String replacer;
+
+					replacer = StringUtils.prependIfMissing(
+							replaceCandidate, "\"");
+					replacer = StringUtils.appendIfMissing(
+							replacer, "\"");
+
+					headerVal = StringUtils.replace(headerVal,
+							replaceCandidate, replacer);
+				}
+
+				responseHeaders.set(headerKey, headerVal);
+				LOG.debug("    > Forwarded");
+			} else {
+				LOG.debug("    > Skipped");
+			}
+		}
+
+		return responseHeaders;
+	}
+
+	/**
+	 * @param ogcMessageDistributor the ogcMessageDistributor to set
+	 */
+	public void setOgcMessageDistributor(OgcMessageDistributor ogcMessageDistributor) {
+		this.ogcMessageDistributor = ogcMessageDistributor;
+	}
+
+
+
+	/**
+	 * @param interceptorRuleService the interceptorRuleService to set
+	 */
+	public void setInterceptorRuleService(
+			InterceptorRuleService<InterceptorRule, ?> interceptorRuleService) {
+		this.interceptorRuleService = interceptorRuleService;
+	}
+
+	/**
+	 * @return the geoServerNameSpaces
+	 */
+	public Properties getGeoServerNameSpaces() {
+		return geoServerNameSpaces;
+	}
+
+	/**
+	 * @param geoServerNameSpaces the geoServerNameSpaces to set
+	 */
+	@Autowired
+	@Qualifier("geoServerNameSpaces")
```

Comment:
In SHOGun2 we are usually using these annotations at the property, but not at the setter. Is there a specific reason? If not, it would be more "uniform" to annotate the property...